includes/Modules/Analytics.php

```diff
@@ -572,6 +563,25 @@ protected function create_data_request( $method, $datapoint, array $data = array
 						}
 						return false;
 					};
+				case 'tag-permission':
+					return function() use ( $data ) {
+						if ( ! isset( $data['tag'] ) ) {
+							/* translators: %s: Missing parameter name */
+							return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'tag' ), array( 'status' => 400 ) );
+						}
+						if ( ! isset( $data['accounts'] ) ) {
+							/* translators: %s: Missing parameter name */
+							return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'accounts' ), array( 'status' => 400 ) );
+						}
+
+						$has_access_to_property = $this->has_access_to_property( $data['tag'], $data['accounts'] );
+						if ( empty( $has_access_to_property ) ) {
+							/* translators: %s: Property id of the existing tag */
+							return new WP_Error( 'google_analytics_existing_tag_permission', sprintf( __( 'We\'ve detected there\'s already an existing Analytics tag on your site (ID %s), but your account doesn\'t seem to have access to this Analytics property. You can either remove the existing tag and connect to a different account, or request access to this property from your team.', 'google-site-kit' ), $data['tag'] ), array( 'status' => 500 ) );
+						}
+
+						return true;
```

Comment:
This piece seems unrelated to the actual change needed as part of the issue (replacing the iframe with a fetch call). It might be a good change, but having it here worries me.

Issues should be scoped to a specific problem, otherwise it's really hard to review and ensure there aren't any regressions (this PR is pretty huge).

Would be great if you could explain how this relates to the iframe -> fetch replacement.

Reply:
@felixarntz I found the verification of user having access to the tag property found was only being done on server side when tag was found via wp_head hook, verification isn't done when tag is found with iframe. I can open a new issue and move this fix in different PR.

Reply:
opened new issue https://github.com/google/site-kit-wp/issues/110